se3cnn/point/kernel.py

```diff
@@ -113,7 +113,10 @@ def forward(self, r):
         kernel = r.new_zeros(batch, self.n_out, self.n_in)
 
         # precompute all needed spherical harmonics
-        Ys = self.sh(self.set_of_l_filters, r)  # [l_filter * m_filter, batch]
+        if r.requires_grad:
+            raise TypeError("The r input of kernel cannot require a gradient.")
```

Comment:
You cannot do that. Someone might want to allow `r` to require grad if he uses its own `sh` function.
Why do you want to do that? Is it the current error message that is not clear enough?